common/src/models/config.rs

```diff
@@ -160,36 +160,45 @@ impl From<SubscriptionOutputFormat> for crate::subscription::SubscriptionOutputF
 }
 
 #[derive(Debug, Clone, Eq, PartialEq, Deserialize)]
-enum PrincsFilterOperation {
+enum ClientFilterOperation {
     Only,
     Except,
 }
 
-impl From<PrincsFilterOperation> for crate::subscription::PrincsFilterOperation {
-    fn from(value: PrincsFilterOperation) -> Self {
+impl From<ClientFilterOperation> for crate::subscription::ClientFilterOperation {
+    fn from(value: ClientFilterOperation) -> Self {
         match value {
-            PrincsFilterOperation::Except => crate::subscription::PrincsFilterOperation::Except,
-            PrincsFilterOperation::Only => crate::subscription::PrincsFilterOperation::Only,
+            ClientFilterOperation::Except => crate::subscription::ClientFilterOperation::Except,
+            ClientFilterOperation::Only => crate::subscription::ClientFilterOperation::Only,
         }
     }
 }
 
 #[derive(Debug, Clone, Eq, PartialEq, Deserialize)]
 #[serde(deny_unknown_fields)]
-struct PrincsFilter {
-    pub operation: Option<PrincsFilterOperation>,
-    pub princs: HashSet<String>,
+struct ClientFilter {
+    pub operation: ClientFilterOperation,
+    #[serde(rename = "type", default)]
+    pub kind: crate::subscription::ClientFilterType,
+    #[serde(default)]
+    pub flags: crate::subscription::ClientFilterFlags,
+    #[serde(default)]
+    pub targets: HashSet<String>,
```

Comment:
I wanted to use an alias, but I had to differentiate the new and the old configuration mode so that I can be backward compatible with case-sensitivity.

Reply:
(same answer as just above: what about keeping case-sensitive filters by default?)